File: ruby/mnam_group.rb
require 'rubygems'
require 'bundler'

module MnamGroup
  def self.init_group group_name, columns
    Mnam.init
  end

  def self.write group_name, key, fields
    Mnam.init

    puts group_name
  end

  def self.write_many group_name, columns
    columns.each{ |key, fields| write group_name, key, fields }
  end

  def self.read group_name, key
    Mnam.init
  end


  private

  def self.init

  end
end

Mnam.write_many 'login', [{'jeff' => [:name => 'jeff']}, {'jeff' => [:name => 'jeff']}, {'jeff' => [:name => 'jeff']}]

File: php/MnamGroup.php
<?php
require 'MnamCass.php';
/**
 * 
 * Mnam: Messaging, not a mess!
 * PHP library port
 * 
 * Class to create/manage message groups
 * 
 * @author  jeffgirard
 * 
 */
class MnamGroup
{
  private static $_MnamCass;
  private $_name;
  private $_mnamFields=array();

  public function __construct($groupName, $fieldNames=null)
  {
    $this->_name = $groupName;

    if($fieldNames && is_array($fieldNames)){
      foreach ($fieldNames as $field) $this->addField($field);
    }
  }

  public function addField($fieldName)
  {
    $this->_mnamFields[] = array(
      'name' => $fieldName,
      'type' => Cassandra::TYPE_UTF8);
  }

  public function commit()
  {
    self::_Init();

    //self::$_MnamCass->getCassandra()->createStandardColumnFamily($keyspace, $groupName, $columns);

    self::$_MnamCass->getCassandra()->createSuperColumnFamily(
        self::$_MnamCass->getKeyspaceName(),
        'cities',
        array(
            array(
                'name' => 'population',
                'type' => Cassandra::TYPE_INTEGER
            ),
            array(
                'name' => 'comment',
                'type' => Cassandra::TYPE_UTF8
            )
        ),
        // see the definition for these additional optional parameters
        Cassandra::TYPE_UTF8,
        Cassandra::TYPE_UTF8,
        Cassandra::TYPE_UTF8,
        'Capitals supercolumn test',
        1000,
        1000,
        0.5
    );
  }

  /**
   * Init a working Mnam instance
   */
  private static function _Init()
  {
    if(!self::$_MnamCass) self::$_MnamCass = new MnamCass();
  }
}

$obj = new MnamGroup('test');
$obj->commit();